src/radix_connect/p2p_links/p2p_link.rs

```diff
@@ -25,6 +27,10 @@ pub struct P2PLink {
     /// for the `ID`.
     pub connection_password: RadixConnectPassword,
 
+    pub connection_purpose: RadixConnectPurpose,
```

Comment:
missing doc, at least a oneliner per field :)

Reply:
e.g. WHO sets the purpose, was this set by DAPP when Sargon / Host wallet app received it. What is the .... purpose of the purpose :D i.e. how does wallet use it.